validator/sawtooth_validator/journal/consensus/dev_mode/dev_mode_consensus.py

```diff
@@ -21,83 +24,140 @@
+        elif self._min_wait_time > 0 and self._max_wait_time <= 0:
+            if self._start_time + self._min_wait_time <= time.time():
+                return True
+        elif self._min_wait_time > 0 \
+                and self._max_wait_time > self._min_wait_time:
+            if self._start_time + self._wait_time <= time.time():
+                return True
+        else:
+            return False
+
+    def finalize_block(self, block):
+        """Finalize a block to be claimed. Provide any signatures and
+        data updates that need to be applied to the block before it is
+        signed and broadcast to the network.
 
         Args:
-            block_header: The candidate block_header to be finalized.
+            journal (Journal): the current journal object
+            block: The candidate block that
         Returns:
-            Boolean: True if the candidate block should be finalized.
+            None
         """
-        block_header.consensus = b"Devmode"
-        return True
+        pass
 
 
 class BlockVerifier(BlockVerifierInterface):
-    """DevMode BlockVerifier implementation
-    """
     def __init__(self, block_cache, state_view):
         super().__init__(block_cache, state_view)
 
         self._block_cache = block_cache
         self._state_view = state_view
 
     def verify_block(self, block_state):
-        return True
+        if block_state.block_header.consensus == b"Devmode":
+            return True
 
 
 class ForkResolver(ForkResolverInterface):
-    """ DevMode ForkResolver implementation
+    """Provides the fork resolution interface for the BlockValidator to use
+    when deciding between 2 forks.
     """
     def __init__(self, block_cache):
         super().__init__(block_cache)
 
         self._block_cache = block_cache
 
+    @staticmethod
+    def hash_signer_pubkey(signer_pubkey, header_signature):
+        m = hashlib.md5()
+        m.update(signer_pubkey)
+        m.update(header_signature)
+        digest = m.hexdigest()
+        number = int(digest, 16)
+        return number
+
     def compare_forks(self, cur_fork_head, new_fork_head):
-        """
+        """The longest chain is selected. If they are equal, then the hash
+        value of the previous block id and publisher signature is computed.
+        The lowest result value is the winning block.
         Args:
-            cur_fork_head (BlockWrapper): The current head of the block chain.
-            new_fork_head (BlockWrapper): The head of the fork that is being
-            evaluated.
+            cur_fork_head: The current head of the block chain.
+            new_fork_head: The head of the fork that is being evaluated.
         Returns:
-            bool: True if the new chain should replace the current chain.
-            False if the new chain should be discarded.
+            bool: True if the result of the signer_pubkey hashed with the
+             header_signature of the new fork head is the lowest value.
```

Comment:
This is only partially true.  You only use the hash if the chains are of equal length.  I think that the generic comment from the ForkResolver interface is sufficient - True if choosing the new chain head, False is choosing the current chain head.